tests/fixtures/application_fixtures.py

```diff
@@ -1671,7 +1671,7 @@ def system_manager(db: Session, system) -> System:
     yield user
     try:
         user.remove_as_system_manager(db, system)
-    except SystemManagerException:
+    except (SystemManagerException, StaleDataError):
```

Comment:
when does a `StaleDataError` occur? And do we need to bypass this err in the actual [endpoint logic](https://github.com/ethyca/fides/pull/2891/files#diff-524e6648deb1d88b996157984be1e98af29b63a75af4943bdeaaac7c0431e319R270) too?

Reply:
This is just for test fixtures - it doesn't need to be in the endpoint.  

The reason is that this test fixture tries to clean itself up after the test by removing the user as a system manager before the user deletes itself.  But this PR adds logic where we remove system managers behind the scenes if a user is downgraded to an approver.  So when the test fixture goes to clean itself up, that data is "stale".

Reply:
If you look at several of the other test fixtures we've had to add logic like this.  where a test it's being used on is testing some sort of delete action that conflicts with the test fixture trying to perform cleanup as a follow up.

Reply:
This makes sense, thanks!